### PR TITLE
Add indexing and search index settings

### PR DESCRIPTION
**Pull request by russcam**
### Description

This PR adds indexing and search settings to index settings. These settings allow configuring of slow logs for indexing and search, as well as setting search idle after.

Search idle after duplicates the individual
`IndexSettings::searchIdleAfter` property, but this pattern is already present e.g. `IndexSettings::blocks` and `IndexSettings::blocksReadOnly`, `IndexSettings::blocksRead`, etc.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by dblock — DISMISSED** [review]
This looks good to me! Do you think you could add some words to the user guide(s) in https://github.com/opensearch-project/opensearch-java/tree/main/guides and maybe a sample to https://github.com/opensearch-project/opensearch-java/tree/main/samples for these features?

File: java-client/src/main/java/org/opensearch/client/opensearch/indices/IndexSettingsIndexing.java [hunk 1]
```diff
@@ -0,0 +1,140 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+/*
```
Comment (reta) [inline java-client/src/main/java/org/opensearch/client/opensearch/indices/IndexSettingsIndexing.java:9]: @russcam could you please remove the `Elasticsearch` license headers in all new files? (unless those where copied from Elasticseach) , thank you

Comment (russcam) [reply java-client/src/main/java/org/opensearch/client/opensearch/indices/IndexSettingsIndexing.java:9]: They're a copy of the following, with modifications to model them better than what is auto-generated:

- [indexing](https://github.com/elastic/elasticsearch-java/blob/main/java-client/src/main/java/co/elastic/clients/elasticsearch/indices/IndexingSlowlogSettings.java#L52)
- [search](https://github.com/elastic/elasticsearch-java/blob/main/java-client/src/main/java/co/elastic/clients/elasticsearch/indices/SettingsSearch.java#L48)

Comment (reta) [reply java-client/src/main/java/org/opensearch/client/opensearch/indices/IndexSettingsIndexing.java:9]: Thanks @russcam , all good than

**Comment by russcam** [comment]
@dblock I've added a sample and docs

**Comment by VachaShah** [comment]
Thank you @russcam! Can you apply `./gradlew :samples:spotlessJavaCheck` to fix the build?

**Comment by russcam** [comment]
@VachaShah done!

**Review by VachaShah — APPROVED** [review]
Thank you @russcam!

**Review by reta — APPROVED** [review]

**Pull request merged**